system/ee/ExpressionEngine/Service/Generator/AddonGenerator.php

```diff
@@ -45,7 +45,8 @@ public function __construct(Filesystem $filesystem, Str $str, array $data)
         $this->str = $str;
 
         $this->name = $data['name'];
-        $this->slug = $this->str->snakecase($data['name']);
+        $this->slug = str_replace(' ', '_', $data['name']);
```

Comment:
I'm wondering if this will handle things like `-` or possibly other names passed in as punctuation. Can we just run `$this->str->snakecase($data['name']);` after processing the slug initially?

Reply:
So I think it will be fine. Since on input it goes through the following
`// Lets filter the name to only allow alphanumerics, "-", "_" and spaces
        $name = preg_replace("/[^A-Za-z0-9 \-_]/", '', $name);`

Reply:
https://github.com/ExpressionEngine/ExpressionEngine/blob/7.dev/system/ee/ExpressionEngine/Cli/Commands/CommandMakeAddon.php#L125

Reply:
Yes, but then we are still allowing `-` in the name. Maybe we want to add something like `$this->slug = str_replace('-', '_', $data['name']);`?

Reply:
Added